nautilus_core/backtest/src/models/fee.rs

```diff
@@ -0,0 +1,152 @@
+// -------------------------------------------------------------------------------------------------
+//  Copyright (C) 2015-2024 Nautech Systems Pty Ltd. All rights reserved.
+//  https://nautechsystems.io
+//
+//  Licensed under the GNU Lesser General Public License Version 3.0 (the "License");
+//  You may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at https://www.gnu.org/licenses/lgpl-3.0.en.html
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------------------------------
+
+use nautilus_model::{
+    instruments::any::InstrumentAny,
+    orders::any::OrderAny,
+    types::{money::Money, price::Price, quantity::Quantity},
+};
+
+pub trait FeeModel {
+    fn get_commission(
+        &self,
+        order: &OrderAny,
+        fill_quantity: Quantity,
+        fill_px: Price,
+        instrument: &InstrumentAny,
+    ) -> Money;
+}
+
+#[derive(Debug, Clone)]
+pub struct FixedFeeModel {
+    commission: Money,
+    zero_commission: Money,
+    change_commission_once: bool,
+}
+
+impl FixedFeeModel {
+    pub fn new(commission: Money, change_commission_once: Option<bool>) -> anyhow::Result<Self> {
+        if commission.as_f64() < 0.0 {
```

Comment:
Something to remember is that Rust is much more efficient with calculations on the domain value objects, so we don't necessary need to cast into ints and doubles as often as the legacy Cython code.